File: Same Tree.c
bool isSameTree(struct TreeNode* p, struct TreeNode* q) {
    if (p == NULL && q == NULL) {
        return true;
    } else if (p == NULL && q != NULL) {
        return false;
    } else if (p != NULL && q == NULL) {
        return false;
    } else  {
        if (p->val == q->val){
            bool lTree=isSameTree(p->left,q->left);
            bool rTree=isSameTree(p->right,q->right);
            return (lTree && rTree);
        }
        else{
            return false;
        }
    }
}

File: Same Tree.js
var isSameTree = function(p, q) {    
    if (p == null && q == null) {
        return true;
    } else if (p == null && q != null) {
        return false;
    } else if (p != null && q == null) {
        return false;
    } else  {
        if (p.val == q.val){
            var lTree=isSameTree(p.left,q.left);
            var rTree=isSameTree(p.right,q.right);
            return (lTree && rTree);
        }
        else{
            return false;
        }
    }
};
